packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -238,10 +257,54 @@ export class Summarizer extends EventEmitter implements ISummarizer {
     }
 
     public readonly summarizeOnDemand: ISummarizer["summarizeOnDemand"] = (...args) => {
-        if (this._disposed || this.runningSummarizer === undefined || this.runningSummarizer.disposed) {
-            throw Error("Summarizer is not running or already disposed.");
+        try {
+            if (this._disposed) {
+                throw Error("Summarizer is already disposed.");
+            }
+            if (this.runtime.summarizerClientId !== undefined &&
+                this.runtime.summarizerClientId !== this.runtime.clientId) {
+                throw Error("On-demand summary attempted while an elected summarizer is present");
+            }
+            const builder = new OnDemandSummarizeResultBuilder();
+            if (this.runningSummarizer && !this.runningSummarizer.disposed) {
+                // Summarizer is already running. Go ahead and start.
+                builder.summarizerStarted.resolve({ success: true, data: undefined });
+                return this.runningSummarizer.summarizeOnDemand(builder, ...args);
+            }
+
+            // Summarizer isn't running, so we need to start it, which is an async operation.
+            // Manage the promise related to creating the cancellation token here.
+            // The promises related to starting, summarizing,
+            // and submitting are communicated to the caller through the results builder.
+            this.runningSummarizer = undefined;
```

Comment:
I feel like (this._disposed) check should be extended to include (this._disposed || this.runningSummarizer?.disposed)
And that will ensure that we always arrive here with this.runningSummarizer === undefined, so no need to set it here.

Reply:
Will do.